### DISCUSSION TITLE
Astro action not showing

### DISCUSSION THREAD

**Discussion by voodoo** [community/community#59005] [Actions] [answered]
### Select Topic Area

Question

### Body

# UPDATE: after I fixed a bug with my code (I had images filtered with .gitignore)  I'm able to deploy with netlify. The workflow item still does not seem to show.

I'm trying to deploy my astro site.  (moving from middleman)

- changed default branch to main
- Added  [.github/workflow/astro](https://github.com/voodoo/voodoo.github.io/blob/main/.github/workflows/astro.yml)
- But it does not show up in https://github.com/voodoo/voodoo.github.io/settings/pages

I'll try to deploy with netlify. I wonder what I did wrong.

**Comment by airtower-luna** [comment]
Your build job fails, the error message looks like some images might be missing: https://github.com/voodoo/voodoo.github.io/actions/runs/5364969484/jobs/9733489963#step:7:25

Linking your repository settings doesn't help, others won't be able to see them. :sweat_smile:

**Reply by voodoo** [reply]
Yes sir, thank you, I see that now. I can deploy with netlify. Still not to gh pages. My workflow file is not showing.  

https://github.com/voodoo/voodoo.github.io/blob/main/.github/workflows/astro.yml

in settings/pages

**Reply by airtower-luna** [reply, marked as answer]
"Sir" is highly inappropriate here, if you want to use a honorific it "ma'am".

Your workflow shouldn't show up in the settings, there you can only choose the deployment source (which should be "custom workflow").

It seems you have deleted the workflow now, but the most recent run says an environment protection rule prevented the deployment: https://github.com/voodoo/voodoo.github.io/actions/runs/5365110077 You'll have to adjust that rule if you want the deployment to work.

**Reply by voodoo** [reply]
Dang it, my apologies airtower-luna, that was dumb. I think I've kinda got it worked out with netlify. Much appreciate it.
